Update stream.py

kylewstew opened the pull request:
Refactored code into InputStream class for clarity and to make code more modular

kylewstew committed:
Update stream.py

Refactored code into InputStream class for clarity and to make code more modular

KayDVC commented on src/data/stream.py:
```diff
@@ -2,116 +2,109 @@
+        finally:
+            self.teardownCameras(cameras)
 
 if __name__ == "__main__":
```
Was used during testing. No longer needed.

KayDVC commented on src/data/stream.py:
```diff
@@ -2,116 +2,109 @@
-
-    teardown(left, right)
-
+class InputStream:
```
Global : Member functions should follow the docstring format.

```
def _create_file_impl(self, f_path: str, f_data) -> bool:
        """The implementation of create file.
        
        Args:
            f_path : the full path to save file data to.
            f_data : the data to save to file.
        Returns:
            True if the operation was a success; false otherwise.
            
        Note:
            If data is of type Numpy Array, the function will use OpenCV's image write function
            function to create a new image.
        """
      ....
```

Include a short description , any arguments passed to the function (if the function takes params), a description of what's returned (if anything is returned), and any major notes for use (as necessary).

KayDVC replied:
The interface will include functionality for output. Please rename to `DataStream`.

KayDVC commented on src/data/stream.py:
```diff
@@ -2,116 +2,109 @@
+        #Tracking for FPS calculation
+        self.old_frame_time = 0
+
+    def setVideoCaptureAttributes(self, cap):
```
Include type hints for returned values. See previous comment for example.

KayDVC commented on src/data/stream.py:
```diff
@@ -2,116 +2,109 @@
+        except:
+            return False
+
+    def setupCameras(self, numCameras):
```
Python standard is snake case for the majority of identifiers including function names. This is not reflected in the design docs due to attempting to separate design from implementation.
``` setupCameras -> setup_camera```

KayDVC commented on src/data/stream.py:
```diff
@@ -2,116 +2,109 @@
+                print(f"Failed to set up camera {i}")
+        return cameras
+
+    def process_frame(self, frame):
```
Any functions not currently in the updated [interface design doc](https://github.com/HelloWorldSeniorProject/AirKeys/tree/develop/documentation/design/output) may be handled elsewhere & are not needed in this interface.

kylewstew committed:
Update stream.py

Added functionality based on design document and updated code based on comments and coding standards.

KayDVC reviewed (approved):
All discussions closed.
